ui/src/hooks/useFetchStatuses.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from "react"
+
+export interface StackStatus {
+  Name: string
+  Error: string
+  Revision: string
+  RepoURL: string
+}
+
+const devData: StackStatus[] = [
+  {
+    Name: "Project Alpha",
+    Error: "",
+    Revision: "v1.0.1",
+    RepoURL: "https://github.com/user/project-alpha"
+  },
+  {
+    Name: "Project Beta",
+    Error: "Failed to build",
+    Revision: "v2.3.4",
+    RepoURL: "https://github.com/user/project-beta"
+  },
+  {
+    Name: "Project Gamma",
+    Error: "",
+    Revision: "v0.9.8",
+    RepoURL: "https://github.com/user/project-gamma"
+  }
+]
+
+async function fetchData(): Promise<StackStatus[]> {
+  const response = await fetch("/stacks")
+  if (!response.ok) {
+    throw new Error("Network response was not ok")
+  }
+
+  return (await response.json()) as StackStatus[]
+}
+
+export default function useFetchStatuses(): {
+  stacks: StackStatus[]
+  error: string | null
```

Comment:
Handle the null case here?

Reply:
Umm what null case? if `error` is null then that means that there was no error and we can access `stacks`. If there is an error then we display it in the calling component

Reply:
Error could just be an empty string if it's empty? I recall checks in the code that first check if it's null then if it's empty, but I could be misremembering

Reply:
Nope, it's only checked for `null`